static/app/views/explore/contexts/spanTagsContext.tsx

```diff
@@ -21,20 +21,21 @@ export const SpanTagsContext = createContext<TypedSpanTags | undefined>(undefine
 interface SpanTagsProviderProps {
   children: React.ReactNode;
   dataset: DiscoverDatasets;
+  enabled: boolean;
 }
 
-export function SpanTagsProvider({children, dataset}: SpanTagsProviderProps) {
+export function SpanTagsProvider({children, dataset, enabled}: SpanTagsProviderProps) {
   const {data: indexedTags} = useSpanFieldCustomTags({
-    enabled: dataset === DiscoverDatasets.SPANS_INDEXED,
+    enabled: dataset === DiscoverDatasets.SPANS_INDEXED && enabled,
```

Comment:
Nit: if you're concerned about undefined being passed through here unexpected, you could always rewrite it as `enabled ? dataset === DiscoverDatasets.SPAN_INDEXED : false` and so on